src/DGtal/shapes/SurfaceMesh.h

```diff
@@ -0,0 +1,773 @@
+
+    /// @return a reference to the vector of normals to vertices.
+    std::vector< RealVector >& vertexNormals() 
+    { return myVertexNormals; }
+
+    /// Mutable accessor to vertex normal.
+    /// @param v any vertex.
+    /// @return the mutable normal associated to \a v.
+    RealVector& vertexNormal( Vertex v )
+    { return myVertexNormals[ v ]; }
+    
+    /// Const accessor to vertex normal.
+    /// @param v any vertex.
+    /// @return the non-mutable normal associated to \a v.
+    const RealVector& vertexNormal( Vertex v ) const
+    { return myVertexNormals[ v ]; }
+    
+    /// @return a const reference to the vector of normals to faces.
+    const std::vector< RealVector >& faceNormals() const
+    { return myFaceNormals; }
+
+    /// @return a reference to the vector of normals to faces.
+    std::vector< RealVector >& faceNormals() 
+    { return myFaceNormals; }
+
+    /// Mutable accessor to face normal.
+    /// @param f any face.
+    /// @return the mutable normal associated to \a f.
+    RealVector& faceNormal( Face f )
+    { return myFaceNormals[ f ]; }
+    
+    /// Const accessor to face normal.
+    /// @param f any face.
+    /// @return the non-mutable normal associated to \a f.
+    const RealVector& faceNormal( Face f ) const
+    { return myFaceNormals[ f ]; }
+    
+    /// @return the average of the length of edges.
+    Scalar averageEdgeLength() const;
+
+    /// @param f any valid face index
+    /// @return the average distance between the centroid of face \a f
+    /// and its vertices.
+    Scalar localWindow( Face f ) const;
+
+    /// Perturbate the positions with a uniform random noise of 'p *
+    /// averageEdgeLength' along arbitrary directions.
+    /// @param p any positive real value.
+    void perturbateWithUniformRandomNoise( Scalar p );
+
+    /// Perturbate the positions with a uniform random noise of 'p *
+    /// averageLocalEdgeLength' along arbitrary directions.
+    /// @param p any positive real value.
+    void perturbateWithAdaptiveUniformRandomNoise( Scalar p );
+
+    /// @param e any valid edge index.
+    /// @return the centroid (or barycenter) of edge \a f.
+    RealPoint edgeCentroid( Index e ) const;
+
+    /// @param f any valid face index.
+    /// @return the centroid (or barycenter) of face \a f.
+    RealPoint faceCentroid( Index f ) const;
+
+    /// @param f any valid face index.
+    /// @return the area of face \a f.
+    Scalar faceArea( Index f ) const;
+
+    /// @param v any valid vertex index.
+    /// @return the Max's weights for each incident face to \a v, in the same order as `myIncidentFaces[ v ]`.
+    /// @note Used in computeVertexNormalsFromFaceNormalsWithMaxWeights
+    Scalars getMaxWeights( Index v ) const;
+    
+    /// Given a ball of radius \a r centered on the centroid of face
+    /// \a f, return the faces having an non empty intersection with
+    /// this ball, each one weighted by its ratio of inclusion.
```

Comment:
```suggestion
    /// this ball, each one weighted by its ratio of inclusion: for edges, the ratio between the length of the part of the edge included in the ball and the total edge length. For faces, the ratio between the @e approximated @e area of the face part inside the ball and the face area).
```

def "inclusion" missing

Reply:
+some details on the approx?

Reply:
Done